engine/pike/pike.go

```diff
@@ -0,0 +1,49 @@
+// Package pike provides functionality to work with Pay-to-PubKey-Hash (P2PKH) scripts in a blockchain context.
+//
+// P2PKH is a common type of Bitcoin transaction that locks funds to a specific public key hash, requiring the
+// corresponding private key to produce a valid signature for spending the funds. This package offers utilities
+// to create, decode, and validate P2PKH scripts.
+//
+// The package includes:
+// - Functions to generate P2PKH addresses from public keys.
+// - Methods to construct P2PKH scriptPubKey and scriptSig scripts.
+// - Utilities to decode and inspect P2PKH scripts and addresses.
+// - Validation functions to ensure the integrity and correctness of P2PKH scripts.
+//
+// This package is intended for developers working with Bitcoin or other cryptocurrencies that support P2PKH transactions.
+// It abstracts the low-level details and provides a high-level interface for creating and handling P2PKH scripts.
+package pike
+
+import (
+	"fmt"
+
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bt/v2/bscript"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/script/template"
+	"github.com/bitcoin-sv/spv-wallet/engine/types/type42"
+)
+
```

Comment:
IMHO we still should have a function `pike.GenerateOutputsTemplate(satoshis)`
which would just wrap call to `template.P2PKH(satoshis)` into a slice

Reply:
done
```suggestion
func GenerateOutputsTemplate(satoshis uint64) ([]template.P2PKHTemplate, error) {
	return template.P2PKH(satoshis)
}
```